src/components/UserSelector.tsx

```diff
@@ -1,42 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { User } from '../types/User';
+import classNames from 'classnames';
+
+interface Props {
+  users: User[];
+  selectedUser: User | null;
+  onSelectUser: (user: User) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selectedUser,
+  onSelectUser,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const handleUserSelect = (user: User) => {
+    onSelectUser(user);
+    setIsDropdownOpen(false);
+  };
+
+  const handleDropdownToggle = () => {
+    setIsDropdownOpen(prev => !prev);
+  };
+
+  const handleOutsideClick = () => {
+    setIsDropdownOpen(false);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
-    <div data-cy="UserSelector" className="dropdown is-active">
+    <div
+      data-cy="UserSelector"
+      className={classNames('dropdown', { 'is-active': isDropdownOpen })}
+    >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={handleDropdownToggle}
+          onBlur={handleOutsideClick}
```

Comment:
Using `onBlur` to close the dropdown might cause it to close unexpectedly when interacting with dropdown items. Consider using a more robust method to detect clicks outside the dropdown, such as attaching a click event listener to the document.